src/java.base/share/classes/java/util/zip/InflaterOutputStream.java

```diff
@@ -71,6 +84,10 @@ private void ensureOpen() throws IOException {
      * Creates a new output stream with a default decompressor and buffer
      * size.
      *
+     * @apiNote
+     * The default decompressor will be closed when this output stream
+     * is {@linkplain #close() closed}.
```

Comment:
This can be normative, meaning you can drop the apiNote.

Reply:
Does the original specification imply a default decompressor is created, or is such a default decompressor simply used?

Reply:
Hello Chen, if you mean whether multiple instances of these streams can share the same default decompressor, then no, they can't (and they don't). Each creates its own instance of default decompressor/compressor.

Reply:
Is it impossible for another compliant implementation of Java SE to do so?

Reply:
It's hard for me to guess what prompted that question, but I'll take a guess - are you asking this question to decide whether we should reword the pre-existing javadoc on these constructors to explicitly state that a new decompressor/compressor instance gets created for each instance of the stream? If so, then my opinion is that we shouldn't change that part of the existing doc.

If you meant something else, then please add the details and we can decide if any additional changes are necessary to the text.

Reply:
The deflater is exposed to subclasses via a protected field so it's part of the API. I think what Jai has is good but I'm nervous about the word "default" as it suggests a global default where as it's a deflater for each output stream.

Reply:
What about something like this?

_An output stream that is created without specifying a decompressor takes responsibility for providing and managing a decompressor, including ensuring that it is properly closed after the output stream is closed. If a decompressor is specified when creating the output stream, the caller takes responsibility for managing that decompressor, including ensuring that it is properly closed after the output stream is closed._